### DISCUSSION TITLE
Nuget restore fails when using custom feed in Azure DevOps

### DISCUSSION THREAD

**Discussion by sf360bojned** [CZEMacLeod/MSBuild.SDK.SystemWeb#74] [Q&A] [answered]
Hi,

We are using this SDK in couple of our web projects, and it works fine in both local and cloud build when using nuget.org as package feed.

We have introduced our own feed and now **nuget restore** step fails with this error:
 error : C:\Program Files\dotnet\sdk\8.0.202\Sdks\MSBuild.SDK.SystemWeb\Sdk not found. Check that a recent enough .NET SDK is installed and/or increase the version specified in global.json.

Version set in global.json is latest (4.0.88).  We have set nuget.config to use our local feed, as well as nuget restore in yml pipeline to use our feed.

Could anyone please advise what could be the problem here?

**Comment by sf360bojned** [comment]
Our feed is fully populated and I have also included MSBuild.SDK.SystemWeb package as well.

**Comment by CZEMacLeod** [comment]
Are you using solely the azure nuget feed or have you got both that and nuget.org listed? Do you have package sources enabled and if so have you made sure they are set correctly?

**Reply by sf360bojned** [reply, marked as answer]
Hi @CZEMacLeod,

After a lot of (unhelpful) nuget  error messages, we pinpointed the issue to our build agents not having permission to access the feed.

I was able to successfully restore and build project when I connected our feed as an external service instead of internal.

Cheers!
